runtime/shared_common/CompositeCache.cpp

```diff
@@ -1711,7 +1711,8 @@ SH_CompositeCacheImpl::startup(J9VMThread* currentThread, J9SharedClassPreinitCo
 	}
 	if (rc == CC_STARTUP_OK) {
 		_started = true;
-		if (NULL == cacheMemory) {
+		/* We don't need to compute the cache id for layer 0. */
```

Comment:
"We don't need to compute the cache id for layer 0" should be:
"We don't need to compute the cache id for the top layer". 
The cache ID is needed only by its higher layer. Top layer does not have a higher layer, so no need to compute cache id for the top layer.

Reply:
Your earlier comments left me with the understanding that a layer stores the unique id of the below it. A layer 0 cache has no layers below it and so doesn't need to compute or store an id.

Reply:
> a layer stores the unique id of the below it. A layer 0 cache has no layers below it

This is correct. 

> so doesn't need to compute or store an id.

The layer 0 does not store an id. But the cache id of layer 0 may needs to be computed. 
If there is a layer 1, the cache unique ID of layer 0 needs to be computed, and stored in layer 1.
If there is no layer 1, layer 0 is the top layer. There is no need to compute or stored id of layer 0.